### PR TITLE
Auth: Use SHA-1 for generating an ID for External Service Role

### PR DESCRIPTION
**Pull request by mgyongyosi**
**What is this feature?**
Instead of using the app/plugin name, use a hash of the name for the external service role's uid, because 'uid' field has a capacity of 40 characters. 

**Why do we need this feature?**
The 'uid' field (of 'role' table) length is 40 characters for pg and mysql. 

**Who is this feature for?**

**Which issue(s) does this PR fix?**:

**Special notes for your reviewer:**
This code does use SHA-1, which has 80 bit collision resistance.

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

**Review by Jguer — APPROVED** [review]

File: pkg/services/accesscontrol/database/externalservices.go [hunk 1]
```diff
@@ -10,9 +12,13 @@ import (
 	"github.com/grafana/grafana/pkg/services/accesscontrol"
 )
 
+// extServiceRoleUID generates a 160 bit unique ID using SHA-1 that fits within the 40 characters limit of the role UID.
 func extServiceRoleUID(externalServiceID string) string {
 	uid := fmt.Sprintf("%s%s_permissions", accesscontrol.ExternalServiceRoleUIDPrefix, externalServiceID)
-	return uid
+	hasher := sha1.New()
+	hasher.Write([]byte(uid))
```
Comment (gamab) [inline pkg/services/accesscontrol/database/externalservices.go:19]: // Ignore gosec G505 as the hash is not used for security purposes
//nolint:gosec
```suggestion
	hasher := sha1.New()
	_, _ = hasher.Write([]byte(uid))
```

Comment (gamab) [reply pkg/services/accesscontrol/database/externalservices.go]: Why not return the error in case there is one?

**Review by gamab — APPROVED** [review]
LGTM :) Left a small comment

File: pkg/services/accesscontrol/database/externalservices.go [hunk 2]
```diff
@@ -10,9 +12,15 @@ import (
...
 )
 
+// extServiceRoleUID generates a 160 bit unique ID using SHA-1 that fits within the 40 characters limit of the role UID.
 func extServiceRoleUID(externalServiceID string) string {
 	uid := fmt.Sprintf("%s%s_permissions", accesscontrol.ExternalServiceRoleUIDPrefix, externalServiceID)
-	return uid
+	// Ignore gosec G505 as the hash is not used for security purposes
+	//nolint:gosec
+	hasher := sha1.New()
+	_, _ = hasher.Write([]byte(uid))
```
[the remark this answers is not in the document]
Comment (mgyongyosi) [reply pkg/services/accesscontrol/database/externalservices.go]: I checked the code (https://github.com/golang/go/blob/master/src/crypto/sha1/sha1.go#L122) and I don’t think it returns an error in any case, so probably the error returh param is there to satisfy the interface of `io.Writer`.

**Pull request merged**